lib/redux/actions/async.js

```diff
@@ -449,21 +464,32 @@ export function setTargetTimezone(userId, timezoneName) {
 
 export function clickDeviceSelectionDone() {
   return (dispatch, getState) => {
-    const { targetDevices, uploadTargetUser } = getState();
+    const { targetDevices, uploadTargetUser, allUsers } = getState();
     const { api } = services;
     dispatch(syncActions.updateProfileRequest());
     if (!_.isEmpty(targetDevices[uploadTargetUser])) {
-      api.user.updateProfile(uploadTargetUser, {
+      let updates = {
         patient: {
           targetDevices: targetDevices[uploadTargetUser]
         }
-      }, (err, profile) => {
+      };
+      api.user.updateProfile(uploadTargetUser, updates, (err, profile) => {
         // suppress unauthorized error until custodial vs normal permissions are ironed out
         // TODO: remove conditional when perms are finalized or accounts are converted
-        if(err && _.get(err,'status') !== 401){
-          dispatch(syncActions.updateProfileFailure(err));
+        if (err) {
+          if (_.get(err,'status') !== 401) {
+            dispatch(syncActions.updateProfileFailure(err));
+          } else {
+            let profile = allUsers[uploadTargetUser];
```

Comment:
Ditto above and would like to see it refactored if, as I suspect, it's common code. Would like to review again to make sure I understand what you're doing in the final version...